tests/ui/collapsible_str_replace.rs

```diff
@@ -0,0 +1,68 @@
+// run-rustfix
+
+#![warn(clippy::collapsible_str_replace)]
+
+fn get_filter() -> &'static str {
+    "u"
+}
+
+fn main() {
+    let misspelled = "hesuo worpd";
+
+    let p = 'p';
+    let s = 's';
+    let u = 'u';
+    let l = "l";
+
+    // LINT CASES
+    let _ = misspelled.replace('s', "l").replace('u', "l");
```

Comment:
IIUC only `char`s in the `from_arg` position get linted now, correct? So not even strings. I think this is an easy extension to also lint strings that can be done in this PR.

I really wouldn't use the `expr_ty` for that, but really the `ExprKind`.

Reply:
Actually, if you [play this playground code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=82605f473fd8895d1013628957753846), I don't think we can lint strings this way. The [pattern](https://doc.rust-lang.org/std/str/pattern/trait.Pattern.html) must be `&[char]`.

I've improved and added tests to handle the following pattern though:

```rust
let _ = "hesudo worpd".replace(d, "l").replace('p', "l").replace("su", "l");
```